libcloud/common/base.py

```diff
@@ -76,6 +76,39 @@
 RETRY_FAILED_HTTP_REQUESTS = False
 
 
+class LazyObject(object):
+    """An object that doesn't get initialized until accessed."""
+
+    @staticmethod
+    def _proxy(lazy_cls, *lazy_init_args, **lazy_init_kwargs):
+        class Proxy(lazy_cls, object):
+            lazy_obj = None
+
+            def __init__(self):
+                pass
+
+            def __getattribute__(self, attr):
+                lazy_obj = object.__getattribute__(self, 'lazy_obj')
```

Comment:
Also, I believe if you override ```__getattr__``` instead of ```__getattribute__``` then you can just check the attribute value without having to invoke ```__getattribute__``` directly like this:

```if self.lazy_obj is None:```